src/os/mac/utf8conv.cpp

```diff
@@ -32,15 +40,68 @@ class Startup {
 static Startup startup;
 
 size_t pws_os::wcstombs(char *dst, size_t maxdstlen,
-                        const wchar_t *src, size_t , bool )
+                        const wchar_t *src, size_t srclen, bool isUTF8)
 {
-  return ::wcstombs(dst, src, maxdstlen) + 1;
+  if (!isUTF8)
+    return ::wcstombs(dst, src, maxdstlen) + 1;
+
+  if (srclen == size_t(-1))
+    srclen = wcslen(src);
+
+  // Convert to UTF-16
+  CFStringRef str = CFStringCreateWithBytes(kCFAllocatorDefault, reinterpret_cast<const unsigned char *>(src), srclen*sizeof(wchar_t), wcharEncoding, false);
+  if (str == NULL)
+    return 0;  // return wcstombs + 1, so 0 to signal error
```

Comment:
What if the input string was of length 0? We could return -1 to distinguish the error case from the empty string case.

Reply:
The logic has not changed here with the previous implementation. If the input string has length 0 then this function will return 1 (see the last statement of the method).

Previously the implementation was simply 'return ::wcstombs(dst, src, maxdstlen) + 1;' so it would also return 1 in that case.